src/main/java/io/xdb/core/state/AsyncStateOptions.java

```diff
@@ -0,0 +1,15 @@
+package io.xdb.core.state;
+
+import com.google.common.base.Strings;
+import io.xdb.core.utils.ProcessUtil;
+import lombok.Builder;
+
+@Builder
+public class AsyncStateOptions {
+
+    private final String id;
+
+    public String getId(final Class<? extends AsyncState> stateClass) {
```

Comment:
similar here, it seems better to pass in the state instance as required for builder?